tests/v2/validation/rbac/globalrolesv2/globalroles_v2_test.go

```diff
@@ -0,0 +1,501 @@
+	require.Equal(gr.T(), expectedRbCount, actualRbCount, "Unexpected number of RoleBindings: Expected %d, Actual %d", expectedRbCount, actualRbCount)
+}
+
+// Create usecases
+func (gr *GlobalRolesV2TestSuite) TestInheritedClusterRoles() {
+	subSession := gr.session.NewSession()
+	defer subSession.Cleanup()
+	gr.createAndTestGlobalRole([]string{roleOwner})
+}
+
+func (gr *GlobalRolesV2TestSuite) TestMultipleInheritedClusterRoles() {
+	subSession := gr.session.NewSession()
+	defer subSession.Cleanup()
+	gr.createAndTestGlobalRole([]string{roleCrtbView, roleProjectsCreate, roleProjectsView})
+}
+
+func (gr *GlobalRolesV2TestSuite) TestInheritedClusterRolesWithCustomClusterRole() {
+	subSession := gr.session.NewSession()
+	defer subSession.Cleanup()
+	log.Info("Create a custom role that inherits rules from the cluster-owner.")
+	inheritedRoleTemplateName := namegen.AppendRandomString("crole")
+	inheritedRoleTemplate, err := gr.client.Management.RoleTemplate.Create(&management.RoleTemplate{
+		Context:         "cluster",
+		Name:            inheritedRoleTemplateName,
+		RoleTemplateIDs: []string{"cluster-owner"},
+	})
+	require.NoError(gr.T(), err)
+
+	gr.createAndTestGlobalRole([]string{inheritedRoleTemplate.ID})
+
+	log.Info("Verify that the user can list all the downstream clusters.")
+	userClient, err := gr.client.AsUser(gr.user)
+	require.NoError(gr.T(), err)
+	clusterList, err := userClient.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+	require.NoError(gr.T(), err)
+	actualClusterCount := len(clusterList.Data)
+	require.Equal(gr.T(), gr.clusterCount, actualClusterCount, "Unexpected number of Clusters: Expected %d, Actual %d", gr.clusterCount, actualClusterCount)
+}
+
+func (gr *GlobalRolesV2TestSuite) TestCreateClustersAfterGlobalRoleAddition() {
+	subSession := gr.session.NewSession()
+	defer subSession.Cleanup()
+	gr.createAndTestGlobalRole([]string{roleMember})
+
+	log.Info("Verify that the user can list all the downstream clusters.")
+	userClient, err := gr.client.AsUser(gr.user)
+	require.NoError(gr.T(), err)
+	clusterList, err := userClient.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+	require.NoError(gr.T(), err)
+	actualClusterCount := len(clusterList.Data)
+	require.Equal(gr.T(), gr.clusterCount, actualClusterCount, "Unexpected number of Clusters: Expected %d, Actual %d", gr.clusterCount, actualClusterCount)
+
+	log.Info("As the new user, create new downstream clusters.")
+	clusterObject, _, testClusterConfig, err := createDownstreamCluster(gr.client, "RKE1")
+	require.NoError(gr.T(), err)
+	provisioning.VerifyRKE1Cluster(gr.T(), gr.client, testClusterConfig, clusterObject)
+
+	_, steveObject, testClusterConfig, err := createDownstreamCluster(gr.client, "RKE2")
+	require.NoError(gr.T(), err)
+	provisioning.VerifyCluster(gr.T(), gr.client, testClusterConfig, steveObject)
+
+	log.Info("Verify that the cluster role template bindings are created for the new downstream cluster.")
```

Comment:
As mentioned earlier, if we do not combine creation and validation of global roles, we can avoid re writing these validations here.